kroxylicious-docs/pom.xml

```diff
@@ -0,0 +1,113 @@
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>kroxylicious-docs</artifactId>
+    <description>Documentation for Kroxylicious</description>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.asciidoctor</groupId>
+                <artifactId>asciidoctor-maven-plugin</artifactId>
+                <version>3.1.1</version>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.asciidoctor</groupId>
+                        <artifactId>asciidoctorj-diagram</artifactId>
+                        <version>2.3.2</version>
+                    </dependency>
+                </dependencies>
+                <executions>
+                    <execution>
+                        <id>convert-kroxylicious-proxy-to-html</id>
+                        <!-- note - that plugin is not bound a phase so won't normally get run -->
+                        <goals>
+                            <goal>process-asciidoc</goal>
+                        </goals>
+                        <configuration>
+                            <requires>
+                                <require>asciidoctor-diagram</require>
+                            </requires>
+                            <sourceDirectory>.</sourceDirectory>
+                            <outputDirectory>${project.build.directory}/docs/html</outputDirectory>
+                            <preserveDirectories>true</preserveDirectories>
+                            <attributes>
+                                <source-highlighter>coderay</source-highlighter>
+                                <imagesdir>./images</imagesdir>
+                                <toc>left</toc>
+                                <icons>font</icons>
+                                <attribute>revnumber!</attribute>
+                                <attribute>revdate!</attribute>
+                                <attribute>revremark!</attribute>
+                            </attributes>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                        <configuration>
+                            <descriptors>
+                                <descriptor>src/main/assembly/asciidoc-distribution.xml</descriptor>
+                            </descriptors>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+    <profiles>
+        <profile>
+            <id>ci</id>
```

Comment:
Yes, we should do this in CI. But I don't think that's the only time we want to generate the docs. We might want to build them locally as developers (and using a profile for that is much more memorable that having to write the whole goal on the command line). Having to say `mvn -Pci` would feel wrong because I'm not CI.

In https://github.com/kroxylicious/kroxylicious/pull/2130 I'm trying to use the existing `dist` profile to decide whether to build the container images. To me the dist should include the docs, so we could do the same here. That means we can say `mvn -Pdist` and build all the pieces which will be in a release.  We might then want to use `-Dci` activate the `dist` profile.

Reply:
Yeah, that would make sense to me.  This was a lift and shift from the parent pom.